### DISCUSSION TITLE
Default Project2ProjectTimeOut value

### DISCUSSION THREAD

**Discussion by fiddlermikey** [NuGet/Home#12554] [Q&A]
What is the default value for -Project2ProjectTimeOut setting when doing a nuget restore? The documentation mentions this option but not the default value

**Comment by zivkan** [comment]
Looks like it's this equation:  https://github.com/NuGet/NuGet.Client/blob/f1757fe7a9a3e178562518f4535f71348a117010/src/NuGet.Clients/NuGet.CommandLine/Commands/RestoreCommand.cs#L673-L674
Where `MSBuildUtility.MsBuildWaitTime` is 2 minutes: https://github.com/NuGet/NuGet.Client/blob/f1757fe7a9a3e178562518f4535f71348a117010/src/NuGet.Clients/NuGet.CommandLine/MsBuildUtility.cs#L24

However, I strongly recommend you stop using nuget.exe, and use [`msbuild -t:restore -p:RestorePackagesConfig=true`](https://learn.microsoft.com/en-us/nuget/reference/msbuild-targets#restoring-packagereference-and-packagesconfig-projects-with-msbuild) instead. You'll never have this msbuild project2project timeout issue again, or have to worry about matching nuget.exe version problems again.
